sql/rpl_rli.cc

```diff
@@ -396,6 +396,39 @@ bool Relay_log_info::mts_finalize_recovery()
   DBUG_RETURN(ret);
 }
 
+bool Relay_log_info::mts_workers_queue_empty()
+{
+  ulong ret= 0;
+
+  for (Slave_worker **it= workers.begin(); it != workers.end(); ++it)
```

Comment:
We don't have to iterate over all the workers if we found at least one with non-zero number of jobs
```suggestion
  for (Slave_worker **it= workers.begin(); ret == 0 && it != workers.end(); ++it)
```